### ISSUE TITLE
[$250] Track – Setting expense to unreported in OldDot turn the expense in workspace chat to 0.00

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#47884]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.24-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4887079&group_by=cases:section_id&group_order=asc&group_id=309130
**Email or phone of affected tester (no customers):** gocemate+a1060@gmail.com
**Issue reported by:**Applause - Internal Team

## Action Performed:
Precondition:
1. In OldDot - Create an unreported transaction on OldDot
2. Verify that it shows up on the self-DM in NewDot
3. In OldDot - add the transaction to a report
4. Verify it is removed from the self-DM in DM and send to Workspace chat
5. In OldDot - make the transaction unreported again

## Expected Result:
Transaction should be removed from workspace chat and send to self-DM chat again

## Actual Result:
Same transaction is present on self-DM chat and workspace chat
Sometimes the transaction in workspace chat appears with amount set to 0.00

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/d8ff2715-3aa8-46fb-a165-fba740c7808e

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018a4221d84dcbfb8b</li>
        <li>Upwork Job ID: 1829267536895122660</li>
        <li>Last Price Increase: 2024-09-05</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@johncschuster  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by Amoralchik** [comment]
I believe it's part of https://github.com/Expensify/App/issues/46287

**Comment by johncschuster** [comment]
Raised in [Slack](https://expensify.slack.com/archives/C06ML6X0W9L/p1724709609931879). Keeping at a lower priority while we focus on higher urgency items.

**Comment by rezkiy37** [comment]
Hi, I am Michael (Mykhailo) from Callstack, an expert agency and I can work on this issue.

**Comment by rezkiy37** [comment]
I am testing the bug.

**Comment by rezkiy37** [comment]
I've completed the testing. I can confirm that it is a backend bug. There are 2 places to fix:
1. Pusher: it does not send updates when the user makes the transaction unreported again. Neither the workspace chat nor the self-DM chat receives updates with added/removed expenses.
2. The backend does not remove unreported transactions from the workspace chat. The backend still includes the transactions and successfully fetches its reports. Also, it duplicates report actions if the user makes the transaction reported again.

We need an internal engineer here 🙂  

https://github.com/user-attachments/assets/68898e80-3a63-4b6e-b825-5bc7f88f65e9

cc @johncschuster

**Comment by johncschuster** [comment]
Amazing. Tanks for doing that, @rezkiy37! @dylanexpensify beat me to it, and has applied the `Internal` label. We'll get someone on this ASAP!

**Comment by johncschuster** [comment]
@dylanexpensify can you help me prioritize this internally?

**Comment by johncschuster** [comment]
Chatted in [Slack](https://expensify.slack.com/archives/C06ML6X0W9L/p1726783792993869?thread_ts=1724709609.931879&cid=C06ML6X0W9L) about this one. We agree this is still worth fixing, but it's currently a lower priority while we focus on more urgent things.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by johncschuster** [comment]
Great! Let's wait one more week. If not reproducible, we can close the issue.

**Comment by johncschuster** [comment]
We need a retest on this, but it looks like we should [wait until the next deploy](https://expensify.slack.com/archives/C9YU7BX5M/p1727946032517319)

**Comment by marcaaron** [comment]
I just retested this and it's very reproducible.

>In OldDot - Create an unreported transaction on OldDot
>Verify that it shows up on the self-DM in NewDot

👍 

>In OldDot - add the transaction to a report

![Screenshot 2024-10-11 at 11 07 01 AM](https://github.com/user-attachments/assets/9e9c1e58-5b26-49b5-a086-94fd24e585b0)

Personal DM `lastMessageText` is wrong. Also, should the Workspace Chat say `tracked $666.00` ? 

When I refresh the page this is what I see...

![Screenshot 2024-10-11 at 11 11 29 AM](https://github.com/user-attachments/assets/1c0b27bd-9dea-4c74-901e-ddf2a34ff818)

**Note:** The expense on the Workspace chat also still says "Unreported" - which is another thing to fix.

![Screenshot 2024-10-11 at 11 11 15 AM](https://github.com/user-attachments/assets/7e060711-8325-471c-8a92-92db4e8277ce)

Which seems correct 👍 

>In OldDot - make the transaction unreported again
>Transaction should be removed from workspace chat and send to self-DM chat again

Yes, well, this doesn't happen. In fact, nothing happens. 

![Screenshot 2024-10-11 at 11 13 04 AM](https://github.com/user-attachments/assets/a3563734-7a2e-4a02-a4f4-e4e8f7b0a178)

But when you refresh the page we get this:

![Screenshot 2024-10-11 at 11 13 54 AM](https://github.com/user-attachments/assets/eb76a2eb-99b7-4789-a8cf-3d4e510401e8)
![Screenshot 2024-10-11 at 11 14 06 AM](https://github.com/user-attachments/assets/0064f9d7-728a-4a5c-8bf7-ed2f890be5b8)

I think what we should do is break this up into a few separate tasks and address each one at a time.

**Comment by marcaaron** [comment]
Oh there's also this other bug...

**Loading state shown incorrectly** 

1. Create a new account in NewDot
2. Create a workspace
3. Sign in on OldDot
4. Create an unreported expense
5. Verify that the expense shows in the self DM in NewDot
💥 we see a loading indicator...

![Screenshot 2024-10-11 at 12 20 43 PM](https://github.com/user-attachments/assets/a0266cab-8bc8-4349-888f-427ddd194f79)

**Why it happens...**

The transaction itself appears to not get sent to the client...

![Screenshot 2024-10-11 at 12 24 50 PM](https://github.com/user-attachments/assets/dc6ec03c-a98b-46ea-b7da-101ea669c69f)

Curious if we have an issue for that yet...

**Comment by marcaaron** [comment]
**Adding unreported transaction to report**

1. Add unreported transaction to report in OldDot
1. Verify that it is removed from the selfDM lastMessageText
1. Verify that it is moved to the Workspace Chat
1. Verify the Workspace Chat text says "tracking $total” and not “tracked”
1. Verify the Workspace shows the correct report preview
1. Verify that the transaction no longer says “Unreported”.

- [ ] TODO

**Removing previously reported transaction by marking it as unreported**

1. Remove reported transaction by unreporting it in OldDot
1. Verify that it is removed from the Workspace Chat and lastMessageText updated correctly
1. Verify that it is moved to the selfDM and lastMessageText updated correctly
1. Verify the Self DM LHN says “tracked $total” and not “tracking $total”
1. Verify the Workspace shows the empty state again
1. Verify that the transaction shows in the Self DM and says “Unreported”

- [ ] TODO

**Comment by marcaaron** [comment]
Checking into some more stuff on the backend for this one... 

It looks we do set the money request report preview with an update like this:

```
{
    "key": "reportActions_1229374452552951",
    "onyxMethod": "merge",
    "value": {
        "5017207174434834948": {
            "message": [
                {
                    "deleted": "2024-10-16 02:00:57.518",
                    "type": "TEXT"
                }
            ],
            "originalMessage": {
                "deleted": "2024-10-16 02:00:57.518"
            }
        }
    }
```

Though, in my local testing the issue persists. The LHN text also remains out of date even though the update appears to get queued from the backend. 

Spent many hours with this. This flow is a bit complex and very broken so it may take some more time to figure it all out.

**Comment by marcaaron** [comment]
One problem appears to be while updates are queued in Auth they are never returned to the web API so that they can be sent in realtime. 

Even if we start queuing the correct updates the user won't see them until they take some action. [Asked in Slack if this is expected](https://expensify.slack.com/archives/C03TQ48KC/p1729134548211619) as I have not worked much on a compatibility issue like this before.

**Comment by marcaaron** [comment]
Ok, I found a bug that was preventing the updates from being sent. We are still needing some additional data to fix this flow however. Some updates look incomplete or broken according to my testing.

**Comment by marcaaron** [comment]
I'll try to pick this up again soon. The effort to fix it is quite high. I think it would have been less if we had considered this during the implementation phase. This is more of a "New Feature" than a "Bug".

**Comment by johncschuster** [comment]
Bumping this for Melv 👍

**Comment by johncschuster** [comment]
@marcaaron can I get a sense check on this one? How's this one going?

**Comment by marcaaron** [comment]
Not great tbh. Very complex and honestly kind of a pain for what we get in return. I think we should make it a Monthly or reconsider whether it needs to be a near term priority.

**Comment by johncschuster** [comment]
No worries! Thanks for the update! I'll go ahead and bump this to `Monthly` right now, and will raise a discussion tomorrow.

**Comment by twisterdotcom** [comment]
I am removing hot pick from any issues with an internal engineer assigned so other engineers can know it's available.

**Comment by marcaaron** [comment]
And I don't think I'm going to be able to finish this one with all the other things on my plate so gonna remove myself to not become a blocker here and will add the label back.

**Comment by johncschuster** [comment]
Looks like we're back to needing an internal engineer to push this one over the line. I'll see if I can rustle someone up on Monday.

**Comment by johncschuster** [comment]
I'm going to retest this behavior to see if it persists, and then figure out how it sits in the greater priority list.

**Comment by johncschuster** [comment]
Whoops! This hasn't been retested. I'll request a retest shortly.

**Comment by johncschuster** [comment]
I couldn't action this today

**Comment by johncschuster** [comment]
This has persisted for a while. I'm asking for a retest before re-rustling an engineer.

**Comment by IuliiaHerets** [comment]
@johncschuster Tester can still repro. The transaction in workspace chat appears with the amount set to 0.00

https://github.com/user-attachments/assets/c7823257-7c14-489f-a74c-2f6fafa735e5

**Comment by johncschuster** [comment]
Sorry I missed your comment above! I action my Dailies every day, but I missed this as it was set to Monthly. I've increased the urgency so I can fit this in tomorrow.

**Comment by johncschuster** [comment]
We're still waiting on an internal engineer to take this on.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)
